karapace/kafka_rest_apis/consumer_manager.py

```diff
@@ -486,7 +489,11 @@ async def fetch(self, internal_name: Tuple[str, str], content_type: str, formats
                 )
                 timeout_left = max(0, (start_time - time.monotonic()) * 1000 + timeout)
                 try:
-                    data = await consumer.getmany(timeout_ms=timeout_left, max_records=1)
+                    message = await consumer.poll(timeout=timeout_left / 1000)
+                    if message is None:
+                        continue
+                    if message.error() is not None:
+                        raise translate_from_kafkaerror(message.error())
```

Comment:
Errors can also surface this way from confluent-kafka, it's a good practice to check for `message.error()`.

However, in case of a topic regex/pattern subscription here we throw an `UnkownTopicOrPartitionError` (subclass of `KafkaError` that is handled below with an HTTP 500) if there is no topic discovered yet that matches the regex.

How fast such a topic gets discovered is dependent on `topic.metadata.refresh.interval.ms`, which is 5 minutes by default. A message with such an error could show up multiple times, thus potentially resulting us dropping (ie. not returning) messages, due to how we immediately raise on exceptions (also later in case of deserialization errors).

Should we perhaps extend the response with an `"error"` in L534 and not throw any exceptions in this case? Also since this error is considered retriable. 🤔